drivers/pwm/pwm_renesas_rx_mtu.c

```diff
@@ -0,0 +1,683 @@
+{
+	const struct pwm_renesas_rx_config *config = dev->config;
+
+	WRITE_BIT(*config->reg.nfcr, config->bit_idx, 1);
+}
+#endif /* CONFIG_PWM_CAPTURE */
+
+static void mtu_start_counter(const struct device *dev)
+{
+	const struct pwm_renesas_rx_config *config = dev->config;
+
+	WRITE_BIT(*config->reg.tstr, config->bit_idx, 1);
+}
+
+static void mtu_stop_counter(const struct device *dev)
+{
+	const struct pwm_renesas_rx_config *config = dev->config;
+
+	WRITE_BIT(*config->reg.tstr, config->bit_idx, 0);
+}
+
+#ifdef CONFIG_PWM_CAPTURE
+static uint16_t mtu_get_counter(const struct device *dev, uint8_t channel)
+{
+	const struct pwm_renesas_rx_config *config = dev->config;
+
+	return config->reg.tgr[channel];
+}
+#endif /* CONFIG_PWM_CAPTURE */
+
+static int pwm_renesas_rx_set_cycles(const struct device *dev, uint32_t channel,
+				     uint32_t period_cycles, uint32_t pulse_cycles,
+				     pwm_flags_t flags)
+{
+	const struct pwm_renesas_rx_config *config = dev->config;
+	uint8_t pwm_state = PWM_STATE_SWITCHING;
+	uint32_t pulse;
+
+	mtu_stop_counter(dev);
+
+	if ((period_cycles > UINT16_MAX) || (pulse_cycles > UINT16_MAX)) {
+		LOG_INF("Fail to set period: %d", period_cycles);
+		return -EINVAL;
+	}
+
+	if (channel >= config->max_num_channels) {
+		LOG_INF("Fail to set channel: %d", channel);
+		return -EINVAL;
+	}
+
+	if (pulse_cycles == period_cycles) {
+		/* 100% duty cycle */
+		if (flags & PWM_POLARITY_INVERTED) {
+			pwm_state = PWM_STATE_0;
+		} else {
+			pwm_state = PWM_STATE_100;
+		}
+
+		/* The PWM device apparently does not change state if pulse_cycles == period_cycles,
+		 * so we have to reduce pulse_cylces by one. Due to the value of pwm_state, the
+		 * signal will remain constant at compare match
+		 */
+		pulse_cycles--;
+	}
+
+	if (pulse_cycles == 0) {
+		/* 0% duty cycle */
+		if (flags & PWM_POLARITY_INVERTED) {
+			pwm_state = PWM_STATE_100;
+		} else {
+			pwm_state = PWM_STATE_0;
+		}
+	}
+
+	/* Enable TOER output when outputting a waveform from the MTIOC pin of MTU3 and MTU4. */
+	mtu_output_enable(dev, channel, 1);
+
+	/* Set the timer I/O control register (TIOR) for a PWM, in this version using PWM mode 1*/
+	config->reg.tior[channel] = pwm_state;
+	pulse = (flags & PWM_POLARITY_INVERTED) ? period_cycles - pulse_cycles : pulse_cycles;
+	config->reg.tgr[channel] = (uint16_t)pulse;
+	pwm_rx_set_period(dev, channel, (uint16_t)period_cycles);
+	*config->reg.tcnt = 0;
+
+	mtu_start_counter(dev);
+
+	return 0;
+}
+
+static int pwm_renesas_rx_get_cycles_per_sec(const struct device *dev, uint32_t channel,
+					     uint64_t *cycles)
+{
+
+	const struct pwm_renesas_rx_config *config = dev->config;
+
+	uint32_t freq_hz;
+	int ret = 0;
+
+	ret = clock_control_get_rate(config->clock, (clock_control_subsys_t)&config->clock_subsys,
+				     &freq_hz);
+
+	switch (config->prescaler) {
+	case RX_MTU_PWM_SOURCE_DIV_1:
+	case RX_MTU_PWM_SOURCE_DIV_4:
+	case RX_MTU_PWM_SOURCE_DIV_16:
+	case RX_MTU_PWM_SOURCE_DIV_64:
```

Comment:
Better to have a `__fallthrough `
```suggestion
	case RX_MTU_PWM_SOURCE_DIV_1:
		__fallthrough 
	case RX_MTU_PWM_SOURCE_DIV_4:
		__fallthrough 
	case RX_MTU_PWM_SOURCE_DIV_16:
		__fallthrough 
	case RX_MTU_PWM_SOURCE_DIV_64:
		__fallthrough 
```

Reply:
Added, thanks